### PR TITLE
feature: allow custom chunking method

### PR DESCRIPTION
**Pull request by rangehow**
In mature RAG frameworks, providing custom chunking methods is a very common choice. Fixed-length splitting brings many disadvantages, such as causing incomplete grammatical inputs, which challenges small models to produce normal outputs. Therefore, we have added an optional feature, allowing users to customize the chunking method, just like they can provide custom LLMs and embedding functions. At the same time, we have provided a separator-based splitting method that can ensure each chunk is grammatically complete.

### CODE REVIEW

File: nano_graphrag/_op.py [hunk 1]
```diff
@@ -48,6 +48,58 @@ def chunking_by_token_size(
...
+):
+    from langchain_text_splitters  import RecursiveCharacterTextSplitter
+    
+
+    text_splitter = RecursiveCharacterTextSplitter(
+        chunk_size=max_token_size,
+        chunk_overlap=overlap_token_size,
+        length_function=lambda x: len(encode_string_by_tiktoken(x)),
+        is_separator_regex=False,
+        # NOTE 这里我没有想好需不需要对外放开一个接口，我感觉修改这块的需求比较微小，可以按照类似PROMPTS自行修改源码的方式处理。
```
Comment (gusye1234) [inline nano_graphrag/_op.py:62]: 我们可以用英文做comments吗
Can we use English for the comment?

File: nano_graphrag/_op.py [hunk 2]
```diff
@@ -48,6 +48,58 @@ def chunking_by_token_size(
         )
     return results
 
+def chunking_by_specific_separators(
```
Comment (gusye1234) [inline nano_graphrag/_op.py:51]: 我推荐把这个function放到examples里面做个例子，感觉没必要为了这个引入langchain_text_splitters的依赖。这个依赖能做很多事情而我们只有当中很简单的一个function。 所以如果依赖是必须的话我还是建议放到examples去。 然后核心代码里面的`nano_graphrag.base`你可以添加chunk func的typing， 比如
`Callable[
    [str, Optional[int], Optional[int]], ...]
]`类似这样的. 有需要的使用的人可以知道如何implement

Comment (rangehow) [reply nano_graphrag/_op.py]: Done.

**Comment by gusye1234** [comment]
LGTM

**Pull request merged**